src/input/input_python.rs

```diff
@@ -203,55 +206,70 @@ impl<'a> Input<'a> for PyAny {
         JsonValue::parse(bytes, true).map_err(|e| map_json_err(self, e))
     }
 
-    fn strict_str(&'a self) -> ValResult<EitherString<'a>> {
-        if let Ok(py_str) = PyString::try_from_exact(self) {
-            Ok(py_str.into())
+    fn validate_str(
+        &'a self,
+        strict: bool,
+        coerce_numbers_to_str: bool,
+    ) -> ValResult<ValidationMatch<EitherString<'a>>> {
+        if let Ok(py_str) = self.downcast_exact::<PyString>() {
+            return Ok(ValidationMatch::exact(py_str.into()));
         } else if let Ok(py_str) = self.downcast::<PyString>() {
             // force to a rust string to make sure behavior is consistent whether or not we go via a
             // rust string in StrConstrainedValidator - e.g. to_lower
-            Ok(py_string_str(py_str)?.into())
-        } else {
-            Err(ValError::new(ErrorTypeDefaults::StringType, self))
+            return Ok(ValidationMatch::strict(py_string_str(py_str)?.into()));
+        }
+
+        'lax: {
```

Comment:
dumb question, what's this syntax?

Reply:
This is called a "label". You can break out of a labelled block with e.g. `break 'lax`. It's quite rare; I use it here because it makes the `return if A { ... } else if B { ... } else { break 'lax; }` chain into an expression, where if any of the `ifs` succeed we'll return. It kept things a little tidier IMO.

Maybe in a world where `strict` is generic there will be a different patter to do this

Reply:
ok